lib/redcord/connection_pool.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+require 'connection_pool'
+require_relative 'redis'
+
+class Redcord::ConnectionPool
+  def initialize(pool_size:, timeout:, **client_options)
+    @connection_pool = ::ConnectionPool.new(size: pool_size, timeout: timeout) do
+      # Construct a new client every time the block gets called
+      Redcord::Redis.new(**client_options, logger: Redcord::Logger.proxy)
```

Comment:
does the redcord logger serialize across threads? Or could the output get interleaved in a non-parsable fashion.

Reply:
I think it may depend on the logger's implementation. Rails logger does handle multi threading.